bitcoin/src/pow.rs

```diff
@@ -248,6 +250,22 @@ impl Target {
     /// The difficulty can only decrease or increase by a factor of 4 max on each difficulty
     /// adjustment period.
     pub fn max_difficulty_transition_threshold(&self) -> Self { Self(self.0 << 2) }
+
+    /// Returns the least number of bits needed to represent the number.
+    pub fn bits(&self) -> u32 { self.0.bits() }
+
+    /// Wrapping multiplication by `u64`.
+    ///
+    /// # Returns
+    ///
+    /// The multiplication result along with a boolean indicating whether an arithmetic overflow
+    /// occurred. If an overflow occurred then the wrapped value is returned.
+    // mutagen false pos mul_u64: replace `|` with `^` (XOR is same as OR when combined with <<)
+    // mutagen false pos mul_u64: replace `|` with `^`
```

Comment:
These comments are out put from `mutagen` runs on the other `mul_u64` function, we shouldn't duplicate them :)

Reply:
Yeah, I just copied and pasted it sorry. I try to ask for clarification in IRC about this mutagen usage. Can you clarify why these feature flags are used?

Reply:
[`mutagen`](https://github.com/llogiq/mutagen) is a mutation testing program that modifies the codebase then runs our unit tests to see if the mutation caused the tests to fail. If the tests do not fail the mutation is said to have survived - indicating we need more unit tests. There are some false positives and these comments are the best way we could come up with to aid future devs running the tests to realize they are false positives.

Reply:
Oh that it is nice to know, thanks